File: src/EvaluateReversePolishNotation.java
class Solution {
    public int evalRPN(String[] tokens) {
        int[] stack = new int[tokens.length];
        int top = -1;
        for (String token : tokens) {
            switch (token) {
                case "+" -> {
                    stack[top - 1] += stack[top];
                    top--;
                }
                case "-" -> {
                    stack[top - 1] -= stack[top];
                    top--;
                }
                case "*" -> {
                    stack[top - 1] *= stack[top];
                    top--;
                }
                case "/" -> {
                    stack[top - 1] /= stack[top];
                    top--;
                }
                default -> stack[++top] = Integer.parseInt(token);
            }
        }
        return stack[top];
    }
}

class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        String[] tokens = {"2", "1", "+", "3", "*"};
        System.out.println(solution.evalRPN(tokens));
    }
}

File: src/EvaluateReversePolishNotation.py
import math

class Solution(object):
    def evalRPN(self, tokens):
        stack = []
        for i in tokens:
            try:
                int(i)
                stack.append(int(i))
            except ValueError:
                a = stack[-2]
                b = stack[-1]
                res = None
                if i == "+":
                    res = a + b
                elif i == "-":
                    res = a - b
                elif i == "*":
                    res = a *  b
                else:
                    res = math.trunc(a / b)
                del stack[-2:]
                stack.append(res)
        return int(stack[-1])


sol = Solution()
print(sol.evalRPN(["10","6","9","3","+","-11","*","/","*","17","+","5","+"]))